httpbakery/form/form.go

```diff
@@ -0,0 +1,151 @@
+// Package form enables interactive login without using a web browser.
+package form
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/juju/httprequest"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/environschema.v1"
+
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+)
+
+/*
+PROTOCOL
+
+A form login works as follows:
+
+	   Client                            Login Service
+	      |                                    |
+	      | GET visitURL with                  |
+	      | "Accept: application/json"         |
+	      |----------------------------------->|
+	      |                                    |
+	      |   Login Methods (including "form") |
+	      |<-----------------------------------|
+	      |                                    |
+	      | GET "form" URL                     |
+	      |----------------------------------->|
+	      |                                    |
+	      |                  Schema definition |
+	      |<-----------------------------------|
+	      |                                    |
+	+-------------+                            |
+	|   Client    |                            |
+	| Interaction |                            |
+	+-------------+                            |
+	      |                                    |
+	      | POST data to "form" URL            |
+	      |----------------------------------->|
+	      |                                    |
+	      |                Form login response |
+	      |<-----------------------------------|
+	      |                                    |
+
+The schema is provided as a environschema.Fileds object. It is the
+client's responsibility to interpret the schema and present it to the
+user.
+*/
+
+// Objects implementing the FormProcessor interface interpret schema and
```

Comment:
I don't like "get" because that to me seems synonymous with collect. "Filler" I like.